src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+/**
+ * The Duke class that runs the Duke program.
+ */
+public class Duke {
+    private static TaskList list;
+    private static FileManager fm;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke.
+     */
+    public Duke() {
+        ui = new Ui();
+        fm = new FileManager();
+        try {
+            list = fm.getListFromFile();
+        } catch (DukeException e) {
+            ui.showError(e);
+            list = new TaskList();
+        }
+    }
+
+    private static void addTask(String[] args) throws DukeException {
+        switch (args[0]) {
+        case "todo":
+            list.add(new Todo(args[1]));
+            break;
+        case "deadline":
+            list.add(new Deadline(args[1], args[2]));
+            break;
+        case "event":
+            list.add(new Event(args[1], args[2]));
+            break;
+        default:
+            break;
+        }
+    }
+
+    /**
+     * A method that runs the program.
+     */
+    public void run() {
```

Comment:
I like the good use of SLAP to ensure that the code is at the same level of abstraction